### ISSUE TITLE
Follow operating system config location standards

### ISSUE THREAD

**Issue by Siilwyn** [vercel/vercel#1276]
It would be great if this module followed the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html). Instead of placing the config directory `.now` in the home directory it should be placed in the OS specific appropriate directory. Getting the correct directory is easy with a module like [env-paths](https://github.com/sindresorhus/env-paths). This, among other things, allows a user to specify a directory in which they want all applications to look for their user configurable setting files and keep their home directory clean.

For more context, this [long standing issue on Atom](https://github.com/atom/atom/issues/8281) contains a lot of replies with people wanting Atom to follow the standards for various reasons.

**Comment by Siilwyn** [comment]
Great to see activity again! Guess you were really busy with v2. :)

Maybe of interest for you, my two cents since I've done this for other projects like [electron-download](https://github.com/electron-userland/electron-download/pull/37), [js-v8flags](https://github.com/js-cli/js-v8flags/pull/38), [greenkeeper](https://github.com/greenkeeperio/rc/pull/3) and [node-gyp](https://github.com/nodejs/node-gyp/pull/1570).

Easiest is making a major breaking change but for user facing config this is a no-go, the now cli falls in this category. There is the option to migrate the config but this can be unexpected behaviour for the user. So the most straightforward way that I would recommend is first reading from the 'old' location and falling back to the new location, documenting this and encouraging users to use the new location.

Code-wise something like this:
```
const path = require('path');
const envPaths = require('env-paths');
const { existsSync } = require('fs');

const paths = envPaths('now', { suffix: '' });

export function getDataDir(): string {
  return maybePath(getOldPath('data')) || paths.data;
}

export function getCacheDir(): string {
  return maybePath(getOldPath('cache')) || paths.cache;
}

export function getConfigDir(): string {
  return maybePath(getOldPath('config')) || paths.config;
}

function maybePath(path) {
  if(existsSync(path)) {
    return path;
  } else {
    return false;
  }
}
```

**Comment by dsifford** [comment]
Just to add to @Siilwyn's comments (which I agree with). You should also begin writing by default to the new location for all new users. That's key. Stop writing to home entirely for new users (unless explicitly asked to do so).

So, in other words, keep looking in the old location first, but stop writing to the old location if nothing exists there already.

**Comment by TooTallNate** [comment]
Closed by #1909.
